lit_tests/kernel/wave/speculative_decoding.py

```diff
@@ -32,21 +32,45 @@ def test_speculative_decoding():
     kernel = wave_compile(options, kernel)
     print(kernel.asm)
 
+    # CHECK: builtin.module {
+    # CHECK-LABEL: func.func @speculative_sampling(
+
+    # Check the function args
+    # CHECK-SAME: %arg0: !stream.binding
+    # CHECK-SAME: %arg3: !stream.binding
+    # CHECK-SAME: %arg13: i32
+
+    # CHECK: arith.constant dense<63> : vector<1xi32>
+
+    # --- Typical operations (representative):
+    # CHECK: gpu.block_id  y upper_bound 10
+    # CHECK: gpu.thread_id  x upper_bound 64
+
+    # CHECK: vector.load
+    # CHECK: vector.store
     # CHECK: scf.while
+    # CHECK: vector.extractelement
     # CHECK: scf.condition
-    # CHECK: scf.while
```

Comment:
why did the scf.while go away from this function?

Reply:
Oh, I restructured it, so I omitted it since it was already matched once above.
Do you suggest including it, or is it okay this way?
I just want to keep it minimal, but still include all representative cases, hence added a few more matches below.

Reply:
Can you keep the while and if to test the structure?